### PR TITLE
feat: add a global http loadbalancer module

### PR DESCRIPTION
**Pull request by muncus**
This is part of exploring how other repositories can consume tf modules from the
RAP platform.

usage sample will follow in a separate repo.

### CODE REVIEW

File: modules/http-loadbalancer-global/main.tf [hunk 1]
```diff
@@ -0,0 +1,86 @@
+/**
```
Comment (stevemcghee) [inline modules/http-loadbalancer-global/main.tf:1]: license

File: modules/http-loadbalancer-global/variables.tf [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+variable "lb_name" {
+  type = string
+  description = "name prefix for generated loadbalancer objects"
+}
+variable "backends" {
+  description = "backends for the load balancer. all fields are needed to identify a kubernetes service."
+  type = map(object({
+    service_obj = any
+    # location = string
```
Comment (stevemcghee) [inline modules/http-loadbalancer-global/variables.tf:13]: is this ok commented like this?

File: modules/http-loadbalancer-global/main.tf [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  locality_lb_policy = "RANDOM"
+  // NEGS go here.
+  // NOTE: zero is not a valid max_rate. must remove whole block to drain.
+  dynamic "backend" {
+    for_each = var.backends
+    content {
+      group = data.google_compute_network_endpoint_group.backend_negs[backend.key].id
+      balancing_mode = "RATE"
+      max_rate_per_endpoint = 100
+      description = format("Kubernetes service %s/%s", backend.value.service_obj.metadata[0].namespace, backend.value.service_obj.metadata[0].name)
```
Comment (stevemcghee) [inline modules/http-loadbalancer-global/main.tf:47]: style: multiline?

File: modules/http-loadbalancer-global/variables.tf (hunk 2 above)
Comment (muncus) [reply modules/http-loadbalancer-global/variables.tf]: oops, prior work. fixed.

**Review by stevemcghee — APPROVED** [review]

**Pull request merged**